.github/workflows/docker-node.yml

```diff
@@ -1,33 +1,40 @@
-name: Prep - Build Docker Images (manual)
+name: Prep - Build Docker Image (on Dockerfile push)
 
+# workflow is for a branch push only and ignores master.
+# push to master (which is also pull request merge) has a more elaborate workflow to run
+# github repo is configured with branch protection on master.
 on:
-  # TODO: revisit.
-  # as of July 2021, there seems to be a GitHub Actions bug causing this workflow to run when a tag is pushed.
-  # adding tags-ignore: causes the workflow to not run at all.
-  # since tag push is used for release trigger, decision was made to build images using manual trigger only.
-
-  # push:
-  #   paths:
-  #     - '.github/docker-node/*.Dockerfile'
+  push:
+    branches-ignore:
+      - 'master'
+    paths:
+      - '.github/docker-node/*.Dockerfile'
 
   workflow_dispatch:
 
 jobs:
   load-docker-node:
+    name: Load Docker Node Config JSON
     runs-on: ubuntu-latest
     outputs:
       matrix: ${{ steps.set-matrix.outputs.matrix }}
+    # github actions triggered by on push will be triggered by both branch and tag push.
+    # the paths filter only applies to branch push and as a result the workflow will run on tag push.
+    # adding tags-ignore: filter causes the workflow to not run at all.
+    # add a conditional to prevent tag push runs.
+    if: ${{ !startsWith(github.ref, 'refs/tags/') }}
```

Comment:
Nice, so this conditional prevents the `load-docker-node` job from running, and since the `build-push` job depends on this one, effectively all jobs are skipped on tag push?

Reply:
Exactly.